lessons/module-3/unit-testing-react.md

```diff
@@ -405,37 +414,33 @@ const Grocery = ({ name, quantity, notes, purchased, starred, onPurchase, onStar
 export default Grocery;
 ```
 
-### Testing the Button Functionality
+### TDD for the Button Functions
 
-So, the buttons say the right things. That's cool, but how do we know that they do 
-the things we expect them to?
+In order to write good tests for buttons, be clear about what should happen when 
+the button is clicked. Notice that `onPurchase`, `onStar`, and `onRemove` properties are being 
+passed in. Reasonably, the onPurchase function will be called when a Purchased button is clicked.
+Similarly, clicking a Star button should call the onStar function.
+Clicking a Delete button should call the onDelete function.
 
-First, we should start by being clear about what should happen. If you look closely 
-at the code, you'll see that `onPurchase`, `onStar`, and `onRemove` properties are being 
-passed in. It stands to reason that these functions should be called when one of those 
-fancy buttons we just made are tested.
+#### Write the test first:
+**Mocks** allow us to confirm a function is being called. Mocks are stubbed-in ("faked") 
+functions that allow us to unit test specific parts of our code where a function is
+being called. A mock will override the behavior of a specific function and provide 
+you with utilities to test the code's interaction with the mock instead.
 
-At this point, we don't necessarily care what the functions are doing -- just that they 
-are being called appropriately. This is where **mocks** come in handy. Mocks are stubbed-in 
-or "faked" functionality that allows us to unit test specific parts of our code without having 
-to worry about others. A mock will override the behavior of a specific function and provide 
-you with utilities to test the interaction with the mock instead.
+When testing applications that use a framework/library like React, making mocks is common and necessary. 
+The general rule here is: if you are not testing the actual behavior **within** the code, then you should be mocking 
+that behavior.
 
-You'll find when testing applications that use a framework like React, you'll need to make 
-a significant amount of mocks. And that's ok! Sometimes it may feel like you're faking too 
-much of your code by mocking so many pieces of functionality. The general rule here is if 
-you're not testing the actual behavior **within** the code you are mocking, it's perfectly 
-fine to mock it.
+Use mocks to test that the functions that were passed in as props are being called appropriately.
 
-So let's use mocks to test that the functions we passed in are being called appropriately.
-
-Consider the following test:
+Add this test:
 
 ```js
 it('should call the onPurchase prop when clicked', () => {
   const onPurchaseMock = jest.fn();
 
-  const wrapper = mount(
+  const wrapper = mount( //mount is good for testing methods on components, see below
```

Comment:
This should just be a `shallow` mount instead

Reply:
Got it.